src/Runner.Worker/JobRunner.cs

```diff
@@ -283,14 +284,21 @@ private async Task<TaskResult> CompleteJobAsync(IRunServer runServer, IExecution
             // Make sure we don't submit secrets as telemetry
             MaskTelemetrySecrets(jobContext.Global.JobTelemetry);
 
+            // Get environment url
+            var environmentUrl = "";
+            if (jobContext?.ActionsEnvironment?.Url is StringToken urlStringToken)
```

Comment:
```suggestion
            if (jobContext.ActionsEnvironment?.Url is StringToken urlStringToken)
```

`jobContext` should never be null